server/routes/api/auth.js

```diff
@@ -62,5 +66,25 @@ router.post('/forgot', async (req, res) => {
   }
 });
 
+// GET Reset Password
+router.get('/reset/:token?', async (req, res) => {
+  const { token } = req.params;
+  if (token === undefined) {
+    return res.status(400).json('Invalid or expired token.');
+  }
+
+  // find user with reset token and check it hasn't expired
+  const user = await User.findOne({
+    'resetPassword.token': req.params.token,
+    'resetPassword.expiration': { $gt: Date.now() },
+  });
+  if (!user) {
+    return res
+      .status(401)
+      .json('Invalid or expired token.')
+      .end();
+  }
+  return res.status(200).json(user);
+});
```

Comment:
So if you actually click on the link from the forgot password email, it will navigate to the correct endpoint. However, it will display all the information from the `user` object in the code block above. I imagine we need to redirect to the forgot password form. I am working on that page here. https://github.com/donthedeveloper/site-builder/pull/117. Not sure how you are supposed to redirect though.